internal/tiltfile/tiltfile_state.go

```diff
@@ -602,6 +602,22 @@ func (s *tiltfileState) assembleK8sV2() error {
 		return err
 	}
 
+	// TODO(dmiller): support all three ways of specifying a name:
```

Comment:
IIUC, if we use something like [`k8sObjectSelector`](https://github.com/windmilleng/tilt/blob/ea3019f1cea5c2d6a03173707377c41db0ac47db/internal/tiltfile/tiltfile_state.go#L1210) (which maybe you don't want to use that specifically because ugh regexes), we don't really need these concepts or to consider these cases separately, and we can cover all the cases with:
1. For each object specified on a resource, construct a selector with as much data is provided.
2. If there is not exactly one k8s object that matches the selector, return an error.

Reply:
Ah this is much better, thank you for pointing me to that